component/discovery/discovery.go

```diff
@@ -53,7 +59,7 @@ func (t *DistributedTargets) Get() []Target {
 			// back to owning the target ourselves.
 			res = append(res, tgt)
 		}
-		if peers[0].Self {
+		if len(peers) >= 1 && peers[0].Self {
```

Comment:
I think this could lead to a bug where the peer list is empty, but in that case we want _someone_ to assign the work to themselves. This is written a little more defensively: 

```suggestion
		if len(peers) == 0 || peers[0].Self {
```